### PR TITLE
Update configure.ac

### PR DESCRIPTION
**Pull request by hdatma**
Allows for curl and xml2 libraries to be retrieved with pkg-config. 
The change is needed under macos, and may be useful elsewhere.

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thank you for submitting a pull request! Maybe @vpodzime can review this?

File: configure.ac [hunk 1]
```diff
@@ -538,26 +538,47 @@ fi
...
-         then
-             AC_MSG_ERROR(Cannot find libcurl library);
-         fi
-      ])
-    AC_CHECK_HEADERS([curl/curl.h],
-      [libcurl_header_found=yes],
-      [if test "x$with_libcurl" != xcheck; then AC_MSG_ERROR(Cannot find libcurl header files); fi])
-  ])
+if [[[ $( which -s pkg-config ) -eq 0 ]]]; then
+   echo "db: pkg-config found";
```
Comment (vpodzime) [inline configure.ac]: Please drop these debug messages.

File: configure.ac [hunk 2]
```diff
@@ -538,26 +538,47 @@ fi
...
+   LIBCURL_CFLAGS="";
+   LIBCURL_CPPFLAGS="$( pkg-config --cflags libcurl )";
+   LIBCURL_LDFLAGS="$( pkg-config --libs libcurl )";
```
Comment (vpodzime) [inline configure.ac]: Can't we use `PKG_CHECK_MODULES`?

File: configure.ac (hunk 1 above)
Comment (hdatma) [reply configure.ac]: Done in p.r. 3500. For some reason, the pull triggered a new number. Please merge.

File: configure.ac (hunk 2 above)
Comment (hdatma) [reply configure.ac]: https://stackoverflow.com/questions/10220946/pkg-check-modules-considered-harmful

**Comment by hdatma** [comment]
Deleting this pull and starting a fresh one, because of problems when rebasing.

**Pull request closed without merging**